src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -244,9 +244,7 @@ def create_keyvault(client, resource_group_name, vault_name, location, #pylint:d
                                         KeyPermissions.restore],
                                   secrets=[SecretPermissions.all],
                                   certificates=[CertificatePermissions.all])
-        object_id = _get_current_user_object_id(graph_client)
-        if not object_id:
-            object_id = _get_object_id(graph_client, subscription=subscription)
+        object_id = _get_object_id(graph_client, subscription=subscription)
```

Comment:
The `_get_object_id` will route to `_get_object_id_by_spn` which still involves a graph query
```python
accounts = list(graph_client.service_principals.list(
        filter="servicePrincipalNames/any(c:c eq '{}')".format(spn)))
```
If my observation is right, I suggest we wrap this line  using `try...catch` and let command to expose a `--object-id` so we don't need to hit the graph and users can still get things done